pkg/kubelet/prober/results/results_manager.go

```diff
@@ -40,14 +40,17 @@ type Manager interface {
 }
 
 // Result is the type for probe results.
-type Result bool
+type Result int
 
 const (
-	// Success is encoded as "true" (type Result)
-	Success Result = true
+	// Success is encoded as 0 (type Result)
+	Success Result = 0
```

Comment:
The startup probe logic is almost the same with liveness probe, right?

Why Success/Failure is sufficient for liveness probe, but not startup probe?

Reply:
OK. Because the init values of liveness probe and startup probes are different.

Reply:
Given that `Unknown` is newly introduced, and only used for initializing startup probe, I think this should be safe.